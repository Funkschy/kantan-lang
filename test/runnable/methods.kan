import "io";

type Struct struct {
    name: string
}

def (s: Struct) method(i: i32) {
    io.printf("%s\n", s.name);
    io.printf("%d\n", i);
}

type Other struct {}

def (s: *Other) method(i: i32) {
    io.printf("%d\n", i);
}

def constructor(): Struct {
    io.printf("...\n");
    return Struct {name: "constructed"};
}

def main() {
    Struct{name: "hello"}.method(1);
    constructor().method(2);
    let s = constructor();
    s.method(3);

    Other{}.method(4);
}
