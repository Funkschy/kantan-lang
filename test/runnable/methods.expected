hello
1
...
constructed
2
...
constructed
3
4
